src/redemptions/redemption.controller.ts

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import {
+  Body,
+  Controller,
+  Get,
+  HttpStatus,
+  NotFoundException,
+  Post,
+  UnprocessableEntityException,
+  UseGuards,
+  ValidationPipe,
+} from '@nestjs/common';
+import { ApiExcludeEndpoint, ApiTags } from '@nestjs/swagger';
+import { MagicLinkGuard } from '../auth/guards/magic-link.guard';
+import { Context } from '../common/decorators/context';
+import { MagicLinkContext } from '../common/interfaces/magic-link-context';
+import { RedemptionDto } from './dto/redemption.dto';
+import { SerializedRedemption } from './interfaces/serializedRedemption';
+import { RedemptionService } from './redemption.service';
+import { serializeRedemption } from './utils/serialize-redemption';
+
+@ApiTags('Redemption')
+@Controller('redemption')
+export class RedemptionController {
+  constructor(private readonly redemptionService: RedemptionService) {}
+
+  @ApiExcludeEndpoint()
+  @Post()
+  @UseGuards(MagicLinkGuard)
+  async create(
+    @Context() { user }: MagicLinkContext,
+    @Body(
+      new ValidationPipe({
+        errorHttpStatusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+      }),
+    )
+    dto: RedemptionDto,
+  ): Promise<SerializedRedemption> {
+    if (await this.redemptionService.find(user)) {
```

Comment:
I think it'd save us some Jumio calls if we check points before creating redemptions for a user. What do you think?

Reply:
This will probably require aggregating points across phases.

Reply:
As discussed offline, we can replace with an eligibility flag on `users`

Reply:
We don't have the data for points yet, but we can add

* Are they from Iran, or North Korea

Reply:
We'll do this in one of the other PRs